bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java

```diff
@@ -66,27 +66,15 @@ class CheckWorkItem extends PullRequestWorkItem {
 
     private final boolean forceUpdate;
     private final boolean spawnedFromIssueBot;
+    private final boolean initialRun;
     private final Map<String, Optional<IssueTrackerIssue>> issues = new HashMap<>();
 
     CheckWorkItem(PullRequestBot bot, String prId, Consumer<RuntimeException> errorHandler, ZonedDateTime triggerUpdatedAt,
-                  boolean needsReadyCheck, boolean forceUpdate) {
-        super(bot, prId, errorHandler, triggerUpdatedAt, needsReadyCheck);
-        this.forceUpdate = forceUpdate;
-        this.spawnedFromIssueBot = false;
-    }
-
-    CheckWorkItem(PullRequestBot bot, String prId, Consumer<RuntimeException> errorHandler, ZonedDateTime triggerUpdatedAt,
-                  boolean needsReadyCheck, boolean forceUpdate, boolean spawnedFromIssueBot) {
+                  boolean needsReadyCheck, boolean forceUpdate, boolean spawnedFromIssueBot, boolean initialRun) {
```

Comment:
Having 4 booleans in the constructor is becoming really hard to read and understand. Perhaps we should introduce factory methods instead so we only allow for valid combinations of the booleans. Something like:
```
fromIssue
fromPr
initialRun
forceUpdate
```

Reply:
I was thinking that if we fix SKARA-2016 in the future, then `forceUpdate` would be removed. So three booleans is reasonable.

Reply:
3 is already a lot, especially if certain combinations aren't actually valid. I was already thinking this constructor was bad before this change. 

When reading the source in intellij, it shows the parameter name at each call site, which alleviates some of the problem, but that isn't happening when reviewing on GitHub, or when merging using other tools.